desktop/docs/electron_architecture.md

```diff
@@ -0,0 +1,114 @@
+## Electron Architecture
+
+### Pinentry
+
+Pinentry will be handled separately from the other desktop MVP features, through
+a new `electron-password` application, which:
+
+1. reads JSON instructions on stdin until EOF
+1. spawns a single window
+1. inside that window, renders HTML from a given path (which can include JS)
+1. prints the user input it received as JSON to stdout
+1. exits
+
+Example stdin:
+```js
+{
+  "html_path": "/usr/local/share/keybase/desktop/templates/password.html",
+  "window_title": "Keybase",
+  "prompt_text": "Please enter your passphrase to unlock your private key",
+  "checkbox": true,
+  "checkbox_text": "Store my passphrase",
+}
```

Comment:
For now, perhaps change `checkbox` to `checkbox_1`, which allows us to extend to multiple checkboxes and radio buttons later.  (Though I hope we'll be able to keep this small.)

Perhaps the `1` should actually correspond to an ID that's present in the supplied HTML?  Then you'd have:
```js
  "checkbox_foo_present": true,
  "checkbox_foo_text": "Store my passphrase"
```
and it would compile down to something like, in JS:
```js
  $("#foo").show()
  $("#foo").text("Store my passphrase")
```
?